### PR TITLE
Improve ci

### PR DESCRIPTION
**Pull request by byronz**
improvement based on 1.1 release:

- make the pip install robust for build jobs
- adding artifacts based on branch and tags,

### CODE REVIEW

File: requires-testing.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-pytest<5.0.0
+pytest
```
Comment (alexcjohnson) [inline requires-testing.txt]: Oh, I didn't notice when <5 came in with https://github.com/plotly/dash/pull/808 - what's the story here? Why did you add it then, why did the restriction get removed now?

Comment (byronz) [reply requires-testing.txt]: There was an issue with pytest at the moment, I was looking for a candidate to trigger a cache key not found, a natural pick for the purpose.
